src/quick-lint-js/parse.h

```diff
@@ -2460,6 +2460,71 @@ class parser {
     v.visit_exit_with_scope();
   }
 
+  bool is_side_effect_free(expression *ast) {
+    switch (ast->kind()) {
+    case expression_kind::_class:
+    case expression_kind::_invalid:
+    case expression_kind::_typeof:
```

Comment:
Blocking: `typeof` might have a side effect, depending on its argument. For example:
```js
typeof 42; // no side effect
typeof f(); // side effect
```